File: src/Golang/26.go
package main

import (
    "fmt"
)

func LengthOfRecurringFactor(n int) int {
    marker := make([]int, n)
    out := 0
    Dividend := 1
    Remainder := 1

    for marker[Dividend%n] == 0 {
        out++
        Remainder = Dividend % n
        Dividend = 10 * Remainder
        fmt.Println(Dividend / n)
        marker[Remainder] = 1
    }

    return out
}

func main() {
    n := 1000 // Example value of n
    max := 0
    result := 0

    for k := 2; k < n; k++ {
        length := LengthOfRecurringFactor(k)
        if max < length {
            max = length
            result = k
        }
    }

    fmt.Printf("Longest recurring cycle is obtained for this integer: %d\n", result)
}

File: src/Ocaml/26.ml
let length_of_recurring_factor n =
  let marker = Array.make n 0 in
  let rec aux dividend out =
    let remainder = dividend mod n in
    if marker.(remainder) = 0 then begin
      marker.(remainder) <- 1;
      let new_dividend = 10 * remainder in
      Printf.printf "%d\n" (new_dividend / n);
      aux new_dividend (out + 1)
    end else
      out
  in
  aux 1 0

let () =
  let n = 1000 in
  let max = ref 0 in
  let result = ref 0 in
  for k = 2 to n - 1 do
    let length = length_of_recurring_factor k in
    if !max < length then begin
      max := length;
      result := k
    end
  done;
  Printf.printf "Longest recurring cycle is obtained for this integer: %d\n" !result
